packages/app-store/paypal/lib/updateAppCredentials.validator.ts

```diff
@@ -0,0 +1,47 @@
+import { z } from "zod";
+
+import type { UpdateAppCredentialsOptions } from "@calcom/trpc/server/routers/viewer/apps/updateAppCredentials.handler";
+
+import Paypal from "./Paypal";
+
+const schema = z.object({
+  credentialId: z.coerce.number(),
+  key: z.object({
+    client_id: z.string(),
+    secret_key: z.string(),
+  }),
+});
+
+const handlePaypalValidations = async ({ input }: UpdateAppCredentialsOptions) => {
+  const validated = schema.safeParse(input);
+  if (!validated.success) throw new Error("Invalid input");
+  const { key } = validated.data;
+
+  // Test credentials before saving
+  const paypalClient = new Paypal({ clientId: key.client_id, secretKey: key.secret_key });
+  const test = await paypalClient.test();
+  if (!test) throw new Error("Provided credentials failed to authenticate");
+
+  // Delete all existing webhooks
+  const webhooksToDelete = await paypalClient.listWebhooks();
+  if (webhooksToDelete) {
+    const promises = webhooksToDelete.map((webhook) => paypalClient.deleteWebhook(webhook));
```

Comment:
I remember I was going to add some kind of filter to this webhooks so we don't delete any existing webhook from the user.